File: clojure/atbash-cipher/src/atbash_cipher.clj
(ns atbash-cipher
  (:require [clojure.string :as str]))

(def cifer "zyxwvutsrqponmlkjihgfedcba")

(defn encode-char [c]
  (let [code (- (int c) 97)]
    (if (<= 0 code 25)
      (.charAt cifer code)
      c)))

(defn encode [plain]
  (->> plain
       str/lower-case
       (re-seq #"[a-z0-9]")
       (apply str)
       (map encode-char)
       (partition-all 5)
       (map str/join)
       (str/join \space)))

File: ruby/atbash-cipher/atbash_cipher.rb
class Atbash
  REVERSE_ALPHABET = ('a'..'z').to_a.reverse.join
  def self.encode(str)
    str.downcase
       .tr('^a-z0-9', '')
       .tr('a-z', REVERSE_ALPHABET)
       .scan(/.{1,5}/)
       .join(' ')
  end

  def self.decode(str)
    str.delete(' ').tr('a-z', REVERSE_ALPHABET)
  end
end
